gradle-baseline-java/src/main/java/com/palantir/baseline/tasks/CheckJUnitDependencies.java

```diff
@@ -49,178 +44,141 @@ public CheckJUnitDependencies() {
 
     @TaskAction
     public final void validateDependencies() {
-        getProbablyTestSourceSets().forEach(ss -> {
-            Optional<Test> maybeTestTask = BaselineTesting.getTestTaskForSourceSet(getProject(), ss);
-            if (!maybeTestTask.isPresent()) {
-                // source set doesn't have a test task, e.g. 'schema'
-                return;
-            }
-            Test task = maybeTestTask.get();
+        getProject()
+                .getExtensions()
+                .getByType(JavaPluginExtension.class)
+                .getSourceSets()
```

Comment:
Calling `.getProject()`/accessing mutable domain state inside task actions is banned for configuration-cache and will throw in some future version of Gradle. Task actions should only access their input/output properties.

It feels weird, but the recommended way to do this is to have the input to the task be just the final result of the check, then the task action does basically nothing other than throw or not. So in this case we could have a `@Optional @Input Property<String> getErrorMessage` here, then just throw the error message if it exists.

Since inputs are only evaluated right before the task runs, it essentially happens at the same time as the task action. But it is configuration cacheable/supports modern gradle.

Reply:
Can we address this in a follow-up? This task was already calling `getProject()` and I'd prefer not to make this PR any larger.